ui/src/main/java/com/xmartlabs/bigbang/ui/common/recyclerview/BaseRecyclerViewAdapter.java

```diff
@@ -0,0 +1,229 @@
+    if (!types.contains(type)) {
+      types.add(type);
+    }
+  }
+
+  @MainThread
+  protected <T extends RecycleItemType> boolean addItems(@NonNull T type, @Nullable List<?> items) {
+    if (CollectionHelper.isNullOrEmpty(items)) {
+      return false;
+    }
+    int lastItemCount = getItemCount();
+    Stream.ofNullable(items)
+        .map(item -> new Element(type, item))
+        .forEach(BaseRecyclerViewAdapter.this.items::add);
+    addTypeIfNeeded(type);
+    if (lastItemCount == 0) {
+      notifyDataSetChanged();
+    } else {
+      notifyItemRangeInserted(Math.max(0, lastItemCount - 1), getItemCount() - lastItemCount);
+    }
+    return true;
+  }
+
+  @MainThread
+  @SuppressWarnings("WeakerAccess")
+  protected <T extends RecycleItemType> void setItems(@NonNull T type, @Nullable List<?> items) {
+    addItems(type, items);
+  }
+
+  /**
+   * Gets all the items count, including dividers
+   *
+   * @return number of total items
+   */
+  @Override
+  public int getItemCount() {
+    int dividerLines = (hasDividers() && getItemCountWithoutDividers() > 0 ? getItemCountWithoutDividers() - 1 : 0);
+    if (hasEndDivider() && getItemCountWithoutDividers() > 0) {
+      dividerLines++;
+    }
+    return getItemCountWithoutDividers() + dividerLines;
+  }
+
+  /**
+   * Gets only the items count from the data
+   *
+   * @return number of total items from the data
+   */
+  @SuppressWarnings("WeakerAccess")
+  protected int getItemCountWithoutDividers() {
+    return items.size();
+  }
+
+  /**
+   * Inflates the view layout/elements.
+   *
+   * @param parent      the parent viewgroup
+   * @param layoutResId the layout resource id
+   * @return the inflated view
+   */
+  protected static View inflateView(@NonNull ViewGroup parent, @LayoutRes int layoutResId) {
+    LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
+    return layoutInflater.inflate(layoutResId, parent, false);
+  }
+
+  /** Binds the divider with the view holder */
+  @SuppressWarnings({"WeakerAccess", "UnusedParameters"})
+  protected void onBindDividerViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
+
+  }
+
+  @CallSuper
+  @Override
+  public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+    if (isADividerPosition(position)) {
+      onBindDividerViewHolder(holder, position);
+    } else {
+      Element element = items.get(getRealItemPosition(position));
+      //noinspection unchecked
+      element.getType().getOnBindViewHolder(holder, element.getItem(), position);
+    }
+  }
+
+  /**
+   * Gets the item position according just to the data items
+   *
+   * @param position of all the items that compose the recycler view
+   * @return item position within data items
+   */
+  @SuppressWarnings("WeakerAccess")
+  protected int getRealItemPosition(int position) {
+    return hasDividers() ? position / 2 : position;
```

Comment:
Please check this math. If the recyclerview has start divider then it doesn't work